cpp/src/parquet/encoding.cc

```diff
@@ -3196,6 +3471,45 @@ class DeltaByteArrayDecoder : public DecoderImpl,
   std::shared_ptr<ResizableBuffer> buffered_data_;
 };
 
+class DeltaByteArrayDecoder : public DeltaByteArrayDecoderImpl<ByteArrayType> {
+ public:
+  using Base = DeltaByteArrayDecoderImpl<ByteArrayType>;
+  using Base::DeltaByteArrayDecoderImpl;
+
+  int Decode(ByteArray* buffer, int max_values) override {
+    return GetInternal(buffer, max_values);
+  }
+};
+
+class DeltaByteArrayFLBADecoder : public DeltaByteArrayDecoderImpl<FLBAType>,
+                                  virtual public FLBADecoder {
+ public:
+  using Base = DeltaByteArrayDecoderImpl<FLBAType>;
+  using Base::DeltaByteArrayDecoderImpl;
+  using Base::pool_;
+
+  int Decode(ByteArray* buffer, int max_values) {
+    return GetInternal(buffer, max_values);
+  }
+  int Decode(FixedLenByteArray* buffer, int max_values) override {
+    int decoded_values_size = max_values;
+    if (MultiplyWithOverflow(decoded_values_size,
+                             descr_->type_length() * sizeof(ByteArray),
+                             &decoded_values_size)) {
+      throw ParquetException("excess expansion in DELTA_LENGTH_BYTE_ARRAY");
+    }
+    ArrowPoolVector<uint8_t> decode_values(decoded_values_size,
+                                           ::arrow::stl::allocator<uint8_t>(pool_));
+    auto decode_buf = reinterpret_cast<ByteArray*>(decode_values.data());
+
+    max_values = GetInternal(decode_buf, max_values);
+    for (int i = 0; i < max_values; i++) {
+      buffer[i].ptr = decode_buf->ptr + i * descr_->type_length();
```

Comment:
Hmm, how is this supposed to work? `decode_buf` is an array of `ByteArray` structures, not some string data. `decode_buf->ptr` is equal to `decode_buf.ptr[0]`. Also this assumes that all strings have the expected length...